merlin/models/tf/transforms/sequence.py

```diff
@@ -62,3 +71,281 @@ def call_outputs(
+
+
+@Block.registry.register_with_multiple_names("seq_predict_next")
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class SeqPredictNext(SequenceTransform):
+    """Prepares sequential inputs and targets for next-item prediction.
+    The target is extracted from the shifted sequence of item ids and
+    the sequential input features are truncated in the last position.
+
+    Parameters
+    ----------
+    schema : Schema
+        The schema with the sequential columns to be truncated
+    target : Union[str, Tags, ColumnSchema]
+        The sequential input column that will be used to extract the target
+    pre : Optional[BlockType], optional
+        A block that is called before this method call().
+        If not set, the ListToRagged() block is applied to convert
+        the tuple representation of sequential features to RaggedTensors,
+        so that the tensors sequences can be shifted/truncated
+    """
+
+    def call(
+        self, inputs: TabularData, targets=None, training=False, testing=False, **kwargs
+    ) -> Union[Prediction, Tuple]:
+        self._check_seq_inputs_targets(inputs)
+
+        # Shifts the target column to be the next item of corresponding input column
+        new_target = inputs[self.target_name][:, 1:]
+        if targets is None:
+            targets = new_target
+        elif isinstance(targets, dict):
+            targets[self.target_name] = new_target
+        else:
+            targets = dict()
+            targets[self.target_name] = new_target
+
+        new_inputs = dict()
+        for k, v in inputs.items():
+            if k in self.schema.column_names:
+                # Removes the last item of the sequence, as it belongs to the target
+                new_inputs[k] = v[:, :-1]
+            else:
+                new_inputs[k] = v
+
+        return (new_inputs, targets)
+
+
+@Block.registry.register_with_multiple_names("seq_predict_last")
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class SeqPredictLast(SequenceTransform):
+    """Prepares sequential inputs and targets for last-item prediction.
+    The target is extracted from the last element of sequence of item ids and
+    the sequential input features are truncated before the last position.
+
+    Parameters
+    ----------
+    schema : Schema
+        The schema with the sequential columns to be truncated
+    target : Union[str, Tags, ColumnSchema]
+        The sequential input column that will be used to extract the target
+    pre : Optional[BlockType], optional
+        A block that is called before this method call().
+        If not set, the ListToRagged() block is applied to convert
+        the tuple representation of sequential features to RaggedTensors,
+        so that the tensors sequences can be processed
+    """
+
+    def call(
+        self, inputs: TabularData, targets=None, training=False, testing=False, **kwargs
+    ) -> Union[Prediction, Tuple]:
```

Comment:
It looks like this will only return `Tuple` in practice. If so we could remove the reference to `Prediction`? Was this here so that this layer could be used inside a model?

Reply:
done